expression/builtin_time.go

```diff
@@ -1770,6 +1770,60 @@ func getTimeZone(ctx context.Context) *time.Location {
 	return ret
 }
 
+//isDuration returns a boolean indicating whether the str is duration
+func isDuration(str string) bool {
+	if n := strings.IndexByte(str, ' '); n >= 0 {
+		if strings.Count(str[:n+1], "-") > 1 {
+			return false
+		}
+		str = str[n+1:]
+	}
+	if n := strings.IndexByte(str, '.'); n >= 0 {
+		str = str[0:n]
+	}
+	seps := strings.Split(str, ":")
+	switch len(seps) {
+	case 1:
+		if length := len(str); length <= 7 && length >= 1 {
+			return true
+		}
+	case 2:
+		return true
+	case 3:
+		return true
+	default:
+		return false
+	}
+	return false
+}
+
+func strDatetimeAddDuration(d string, arg1 types.Duration) (result types.Datum, err error) {
+	arg0, err := types.ParseTime(d, mysql.TypeDatetime, getFsp(d))
+	if err != nil {
+		return result, err
+	}
+	tmpDuration := arg0.Add(arg1)
+	resultDuration, err := tmpDuration.ConvertToTime(mysql.TypeDatetime)
+	if err != nil {
+		return result, err
+	}
+	result.SetString(resultDuration.String())
+	return result, nil
```

Comment:
I do not fix `return`.if do it ,i will get the compiler error ‘err is shadowed during return’